src/crepe/ComponentManager.h

```diff
@@ -43,6 +39,11 @@ class ComponentManager {
 	 */
 	template <typename T, typename... Args>
 	T & add_component(game_object_id_t id, Args &&... args);
+	//! GameObject is used as an interface to add components instead of the
+	// component manager directly
+	friend class GameObject;
+
+public:
 	/**
 	 * \brief Delete all components of a specific type and id
```

Comment:
Dit is niet voor nu, maar voor later: Misschien is het een goed idee om alle `delete_components_...` methodes protected/private te maken en de `SceneManager` als friend toe te voegen (zodat alleen de `SceneManager` components kan deleten).